vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDeviceProvider.kt

```diff
@@ -167,7 +167,7 @@ class AndroidDeviceProvider : DeviceProvider, CoroutineScope {
     private fun matchDdmsToDevice(device: IDevice): AndroidDevice? {
         val observedDevices = devices.values
         return observedDevices.findLast {
```

Comment:
with devices being a `ConcurrentHashMap` instance, shouldn't this use some concurrent retrieval method instead?